### PR TITLE
fix(async.pool): making logging optional

### PR DESCRIPTION
**Pull request by kadukf-outreach**
## What this PR does / why we need it
Logging in the async.pool is noisy, errors can be a business values handled in the business code (moving/extracting the code)

## Jira ID

[XX-XX]

### CODE REVIEW

File: pkg/async/pool/pool.go [hunk 1]
```diff
@@ -273,3 +262,27 @@ type unit struct {
 	Context context.Context
 	Runner  async.Runner
 }
+
+type loggingScheduler struct {
+	Inner Scheduler
+	Name  string
```
Comment (dmytrokasianenko-outreach) [inline pkg/async/pool/pool.go:268]: there is already option for Name, would it make sense to remove it?

Comment (kadukf-outreach) [reply pkg/async/pool/pool.go:268]: removing it

**Review by dmytrokasianenko-outreach — APPROVED** [review]

File: pkg/async/pool/pool.go [hunk 2]
```diff
@@ -273,3 +255,27 @@ type unit struct {
...
+}
+
+func (w *loggingScheduler) Schedule(ctx context.Context, r async.Runner) error {
+	return w.log(ctx, w.Inner.Schedule(ctx, async.Func(func(ctx context.Context) error {
+		return w.log(ctx, r.Run(ctx))
+	})))
+}
+
+// WithLogging creates a scheduler which logs the errors returned from the scheduling as well as executing phase
```
Comment (artemchernyak-outreach) [inline pkg/async/pool/pool.go:270]: Does this mean that if a service update gobox without switching over to `WithLogging` they will lose logging without warning?

If that's the case, I think that we should make this `WithoutLogging` and make the default include logging.

Comment (vk-outreach) [reply pkg/async/pool/pool.go:270]: Yeah, looks like this is in use only in mailroomapi and ppsgateway.  So, if you are willing to manually upgrade those repos, then this is ok.

**Review by artemchernyak-outreach — APPROVED** [review]

**Review by vk-outreach — APPROVED** [review]
LGTM given that the breaking change is limited to only a couple of repos.

A slightly different approach is to remove this option altogether and instead export two helpers:

```golang
func LogScheduler(s Scheduler, args ...log.Marshaler) Scheduler {
    ....  just export loggingScheduler
}

func LogRunner(r async.Runner, args ...log.Marshaler) async.Runner {
    ... just export a new runner which also logs on errors
}
```

I am ok with this specific change though.

File: pkg/async/pool/pool.go (hunk 2 above)
Comment (artemchernyak-outreach) [reply pkg/async/pool/pool.go:270]: Sounds good to me. Thanks for looking into it!

**Review by artemchernyak-outreach — APPROVED** [review]

**Pull request merged**